php/Blockonomics.php

```diff
@@ -79,20 +111,113 @@ public function get_price($currency, $crypto)
         return $responseObj;
     }
 
-    public function update_callback($callback_url, $xpub)
+    public function update_callback($callback_url, $crypto, $xpub)
     {
-        $url = Blockonomics::SET_CALLBACK_URL;
+        if ($crypto == 'btc'){
+            $url = Blockonomics::SET_CALLBACK_URL;
+        }else{
+            $url = Blockonomics::BCH_SET_CALLBACK_URL;
+        }
         $body = json_encode(array('callback' => $callback_url, 'xpub' => $xpub));
         $response = $this->post($url, $this->api_key, $body);
         return json_decode(wp_remote_retrieve_body($response));
     }
 
-    public function get_callbacks()
+    public function get_callbacks($crypto)
     {
-        $url = Blockonomics::GET_CALLBACKS_URL;
+        if ($crypto == 'btc'){
+            $url = Blockonomics::GET_CALLBACKS_URL;
+        }else{
+            $url = Blockonomics::BCH_GET_CALLBACKS_URL;
+        }
         $response = $this->get($url, $this->api_key);
         return $response;
     }
+    
+    public function check_get_callbacks_response($response, $response_body, $crypto){
+        $error_str = '';
+        $error_crypto = strtoupper($crypto).' error: ';
+        //TODO: Check This: WE should actually check code for timeout
+        if (!wp_remote_retrieve_response_code($response)) {
+            $error_str = __($error_crypto.'Your server is blocking outgoing HTTPS calls', 'blockonomics-bitcoin-payments');
+        }
+        elseif (wp_remote_retrieve_response_code($response)==401)
+            $error_str = __($error_crypto.'API Key is incorrect', 'blockonomics-bitcoin-payments');
+        elseif (wp_remote_retrieve_response_code($response)!=200)
+            $error_str = $error_crypto.$response->data;
+        elseif (!isset($response_body) || count($response_body) == 0)
```

Comment:
Checking the response body can be moved into set_callback. This way check_get_callbacks_response can handle checking response codes `wp_remote_retrieve_response_code($response)` and set_callback can handle checking the response body `json_decode(wp_remote_retrieve_body($response))`. 
Let's also rename functions `check_get_callbacks_response` to `check_get_callbacks_response_code($response, $crypto)` and `set_callback` to `check_get_callbacks_response_body($response, $crypto)`.

Reply:
I've made the changes and committed them. At the moment it's not too clear where the callback is being set when examining `check_callback_urls_or_set_one`. I was thinking that we could try moving out `update_callback` from `check_get_callbacks_response_body`. However, it seems like quite a messy process since it's dependent on quite many variables. Do you have any suggestions?